### DISCUSSION TITLE
Transfer learning on cifar-10

### DISCUSSION THREAD

**Discussion by SiebeF** [deepjavalibrary/djl#1775] [Q&A]
Hello, I'm a beginner in DJL. I followed the transferlearning tutorial with cifar10. So now I trained the model, but I really can't seem to figure out how to test it. The tutorial says follow the inference tutorial, but I can't figure it out. Hopefully somebody can help me, thanks :)

**Comment by frankfliu** [comment]
You can find the code in the transfer learning example code, it shows you how to run inference with your saved parameters file:

https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/training/transferlearning/TrainResnetWithCifar10.java#L185-L212

**Reply by SiebeF** [reply]
Thanks, but that was the part where I was struggling, I don't know how to call this function correctly in main. I understand how to define the path, but I don't know the other parts. Maybe somebody has an example main code calling the function for the already trained model, I don't need to train it again?

**Comment by frankfliu** [comment]
@SiebeF 
Hope the following can gave you some idea:
```
    public static void main(String[] args) throws IOException, ModelException, TranslateException {
        Model model = Model.newInstance("resnetv1");
        Block resNet50 =
                ResNetV1.builder()
                        .setImageShape(new Shape(3, 32, 32))
                        .setNumLayers(50)
                        .setOutSize(10)
                        .build();
        model.setBlock(resNet50);

        // load the trained model parameters
        Path modelPath = Paths.get("build/model");
        model.load(modelPath);

        String synsetUrl =
                "https://mlrepo.djl.ai/model/cv/image_classification/ai/djl/mxnet/synset_cifar10.txt";
        ImageClassificationTranslator translator =
                ImageClassificationTranslator.builder()
                        .addTransform(new ToTensor())
                        .addTransform(new Normalize(Cifar10.NORMALIZE_MEAN, Cifar10.NORMALIZE_STD))
                        .optSynsetUrl(synsetUrl)
                        .optApplySoftmax(true)
                        .build();

        Image img = ImageFactory.getInstance().fromUrl("src/test/resources/airplane1.png");

        try (Predictor<Image, Classifications> predictor = model.newPredictor(translator)) {
            System.out.println(predictor.predict(img));
        }
    }
```
